src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeResource.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.barcode
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseBody
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.config.ErrorResponse
+import javax.validation.constraints.NotEmpty
+import javax.validation.constraints.NotNull
+
+@RestController
+@RequestMapping(produces = [MediaType.APPLICATION_JSON_VALUE])
+class BarcodeResource(private val barcodeService: BarcodeService) {
+
+  @PostMapping(value = ["/barcode"])
+  @ResponseBody
+  // TODO this will be protected by a role
+  // @PreAuthorize("hasRole('ROLE_CREATE_BARCODE')")
+  @Operation(
+    summary = "Creates a one time barcode for the prisoner",
+  )
+  @ApiResponses(
```

Comment:
Yes it's the local convention. It probably makes more sense when writing an API with multiple consumers rather than the BFF pattern we have... but there are benefits you mention, one of which is used to generate the types in the Typescript project.

I would be interested in using it for some tests (I think there are some projects using the swagger docs in PACT test?) though our API will be very small it's probably overkill in our case.